### PR TITLE
Mobile party properties

### LINKED ISSUE #783: Sync MobileParty Properties
## Description

1. Sync the values in the **Synced** list below.
2. Create an integration test for all values (1 for server changing the value and 1 for client changing the value if applicable)
3. Create a command to test changing the value in game [see similar command](https://github.com/Bannerlord-Coop-Team/BannerlordCoop/blob/development/source/GameInterface/Services/Towns/Commands/TownDebugCommand.cs#L455)

Also see additional information at the bottom for more information and templates

**Key**
Server side only (only allow the running of the function on the server side)
Client side only (only allow the running of the function on the client side)
Client side sync (request the server to change the value, server then changes the value and replies to the client allowing the change)
Server side sync (only server allows running of the function and send to all clients the value changed and clients update the value/call original function)

**Synced**

| Method/Field Name | Sync Type |
|:-----------------|:---------:|
| Army | Server side only |
| CustomName | Server side only |
| LastVisitedSettlement | Server side only |
| Aggressiveness | Server side only |
| ArmyPositionAdder | Server side only |
| Objective | Server side only |
| Ai | Server side only |
| Party | Server side only |
| IsActive | Server side only |
| ThinkParamsCache | Server side only |
| ShortTermBehavior | Server side only |
| IsPartyTradeActive | Server side only |
| PartyTradeGold | Server side only |
| PartyTradeTaxGold | Server side only |
| StationaryStartTime | Server side only |
| VersionNo | Server side only |
| ShouldJoinPlayerBattles | Server side only |
| IsDisbanding | Server side only |
| CurrentSettlement | Server side only |
| AttachedTo | Server side only |
| BesiegerCamp | Server side only |
| Scout | Server side only |
| Engineer | Server side only |
| Quartermaster | Server side only |
| Surgeon | Server side only |
| ActualClan | Server side only |
| RecentEventsMorale | Server side only |
| IsInspected | Server side only |
| MapEventSide | Server side only |
| PartyComponent | Server side only |
| IsMilitia | Server side only |
| IsLordParty | Server side only |
| IsVillager | Server side only |
| IsCaravan | Server side only |
| IsGarrison | Server side only |
| IsCustomParty | Server side only |
| IsBandit | Server side only |

**Deferred**

None

**Externally Deferred**

None

**Non-Synced**
| Method/Field Name | Rational |
|:-----------------|:---------|
| EventPositionAdder | This is a position on the map and can be simulated on all clients/server | 

## Intended Design

You are expected to only sync the property setters.
Try using the new [autosync tool](https://github.com/Bannerlord-Coop-Team/BannerlordCoop/wiki/Property-AutoSync)

## Location

Create a branch based from [development](https://github.com/Bannerlord-Coop-Team/BannerlordCoop)

## Related Issues

N/A

## Requirements

N/A

## Additional information

Using the [GameInterface Service Tempale](https://github.com/Bannerlord-Coop-Team/BannerlordCoop/tree/development/source/GameInterface/Services/Template) create commands to test the synced values.

For network communication use [Server Service Template](https://github.com/Bannerlord-Coop-Team/BannerlordCoop/tree/development/source/Coop.Core/Server/Services/Template) and [Client Service Template](https://github.com/Bannerlord-Coop-Team/BannerlordCoop/tree/development/source/Coop.Core/Client/Services/Template)

It is recommended to start with an integration test, you can create one using the [test template](https://github.com/Bannerlord-Coop-Team/BannerlordCoop/tree/development/source/Coop.IntegrationTests/Template).

## Definition of Done
- [ ] Class level comments exist for all new classes.
- [ ] XUnit tests exist for every method that does not require the game to be ran.

### PR DESCRIPTION
**Pull request by EgardA**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] All new classes have class-level documentation comments, if there are any at all
- [x] Tests for the changes have been added (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation update
- [ ] Other... Please describe:

## What is the new behavior?

Resolves #783

### CODE REVIEW

File: source/GameInterface/Services/MobileParties/Patches/PartyTickPatch.cs [hunk 1]
```diff
@@ -9,6 +9,6 @@ namespace GameInterface.Services.MobileParties.Patches;
 [HarmonyPatch(typeof(CampaignPeriodicEventManager))]
 internal class PartyTickPatch
 {
-    [HarmonyPatch(nameof(CampaignPeriodicEventManager.MobilePartyHourlyTick))]
+    [HarmonyPatch(nameof(CampaignPeriodicEventManager.TickPeriodicEvents))]
```
Comment (garrettluskey) [inline source/GameInterface/Services/MobileParties/Patches/PartyTickPatch.cs]: MobilePartyHourlyTick still should only be server side

File: source/GameInterface.Tests/Bootstrap/Patches/MapScenePatches.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+[HarmonyPatch(typeof(MapScene))]
+internal class MapScenePatches
+{
+    [HarmonyPatch(nameof(MapScene.GetFaceTerrainType))]
+    [HarmonyPrefix]
+    static bool GetFaceTerrainTypePrefix(ref TerrainType __result)
+    {
+        Array values = Enum.GetValues(typeof(TerrainType));
+        Random random = new Random();
```
Comment (EgardA) [inline source/GameInterface.Tests/Bootstrap/Patches/MapScenePatches.cs:17]: Is there a reason to use Random here? Wouldn't this create different terrain types for every player?

File: source/GameInterface.Tests/Bootstrap/Patches/DefaultMapWeatherModel.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+[HarmonyPatch(typeof(DefaultMapWeatherModel))]
+internal class DefaultMapWeatherModelPatches
+{
+
+    [HarmonyPatch(nameof(DefaultMapWeatherModel.GetWeatherEventInPosition))]
+    [HarmonyPrefix]
+    static bool GetWeatherEventInPositionPrefix(ref WeatherEvent __result)
+    {
+        Array values = Enum.GetValues(typeof(WeatherEvent));
+        Random random = new Random();
```
Comment (EgardA) [inline source/GameInterface.Tests/Bootstrap/Patches/DefaultMapWeatherModel.cs:20]: Is there a reason to use Random here? Wouldn't this create different weather types for every player?

**Review by EgardA — COMMENTED** [review]
Just two questions,

I cannot approve the PR as I am the creator of it.

File: source/GameInterface.Tests/Bootstrap/Patches/MapScenePatches.cs (hunk 2 above)
Comment (garrettluskey) [reply source/GameInterface.Tests/Bootstrap/Patches/MapScenePatches.cs:17]: This is for testing only, these patches are located in GameInterface.Tests

File: source/GameInterface.Tests/Bootstrap/Patches/DefaultMapWeatherModel.cs (hunk 3 above)
Comment (garrettluskey) [reply source/GameInterface.Tests/Bootstrap/Patches/DefaultMapWeatherModel.cs:20]: This is for testing only, these patches are located in GameInterface.Tests

**Review by garrettluskey — CHANGES_REQUESTED** [review]

File: source/GameInterface/Services/MobileParties/Patches/PartyTickPatch.cs (hunk 1 above)
Comment (garrettluskey) [reply source/GameInterface/Services/MobileParties/Patches/PartyTickPatch.cs]: Fixed

**Review by garrettluskey — DISMISSED** [review]

File: source/GameInterface.Tests/Bootstrap/Patches/DefaultMapWeatherModel.cs (hunk 3 above)
Comment (EgardA) [reply source/GameInterface.Tests/Bootstrap/Patches/DefaultMapWeatherModel.cs:20]: Haha oppps, missed that.

File: source/GameInterface.Tests/Bootstrap/Patches/MapScenePatches.cs (hunk 2 above)
Comment (EgardA) [reply source/GameInterface.Tests/Bootstrap/Patches/MapScenePatches.cs:17]: Haha oppps, missed that.

**Pull request merged**